contributors/design-proposals/propagation.md

```diff
@@ -142,13 +153,41 @@ distros.
 1. (From @euank) Changing those mountflags may make docker even less stable,
 this may lock up kernel accidentally or potentially leak mounts.
 
+1. (From @jsafrane) Typical container that needs to mount something needs to
+see host's `/dev` and `/sys` as HostPath volumes. This would make them shared
+without any way to opt-out. Docker creates a new `/dev/shm` in the
+container, which gets propagated to the host, shadowing host's `/dev/shm`.
+Similarly, systemd running in a container is very picky about `/sys/fs/cgroup`
+and something prevents it from starting if `/sys` is shared.
 
 ## Decision
 
-We will take 'Make HostPath shared for privileged containers, slave for
-non-privileged', an environment check and an WARNING log will be emitted about
-whether propagation mode is supported.
-
+* We will take 'Add an option in VolumeMount API' (with an annotation during
```

Comment:
I disagree with this decision. I think this should be added to the HostPathVolumeSource, since it only applies to HostPath volumes. I think it's a relatively small edge case that there are multiple containers in the same pod mounting the same HostPath volume, where one wants it to be shared and the other wants slave. In that case, if one of the containers was not privileged, it would get the non-shared mount anyway.